docs/reference/files/file-libname-py.md

```diff
@@ -0,0 +1,160 @@
+
+The docstring supports Markdown, specifically, CommonMark. 
+
+
+### Metadata
+
+After the docstring, there are a few metadata keys, as below.
+
+
+#### `LIBID`
+
+The `LIBID` key controls the unique identifier for a library across the entire universe of charms. The value is assigned by Charmhub to this particular library automatically at library creation time. This key enables Charmhub and `charmcraft` to track the library uniquely even if the charm or the library are renamed, which allows updates to warn and guide users through the process.
+
+#### `LIBAPI`
+
+`LIBAPI` is set to an initial state of `0`. In general, `LIBAPI` must match the major version in the import path.
+
+#### `LIBPATCH`
+
+`LIBPATCH` is set to an initial state of `1`. In general, it must match the current patch version (needs to be updated when changing). 
+
+#### `PYDEPS`
+
+The `PYDEPS` key is an optional key that allows you to declare external Python dependencies. Charmcraft will make sure to add them to the dependencies required for your charm. 
+
+The key maps to a list of strings. Each string is a regular "pip installable" Python dependency that will be retrieved from PyPI in the usual way (subject to the user's system configuration) and which supports all dependency formats (just the package name, a link to a Github project, etc.).
+
+Some examples of possible PYDEPS declarations are as below:
+
+```
+PYDEPS=["jinja2"]
+
+PYDEPS = ["pyyaml", "httpcore<0.15.0,>=0.14.5"]
+
+PYDEPS = [
+    "git+https://github.com/canonical/operator/#egg=ops",
+    "httpcore<0.15.0,>=0.14.5",
+    "requests",
+]
+```
+
+Of course, only one `PYDEPS` declaration is allowed.
+
+Charmcraft will collect the dependencies from all libraries included in the charm and install them from source in the virtual environment inside the `.charm` file, together with all the other Python dependencies declared by the charm itself.
+
+Note that, when called to install all the dependencies from the charm and all the used libraries, `pip` may detect conflicts between the requested packages and their versions. This is a feature, because it's always better to detect incompatibilities between dependencies at this moment than when the charm is being deployed or run after deployment.
+
+
+### Code
+
+After the docstring and the metadata, there's the library code. This is regular Python code.
+
+## Popular library index
+
+
+This is an index of the charm libraries that are currently known to be available.
+
+```{note}
+
+This list may be missing libraries from charms that are not publicly listed on Charmhub. If you would like to add a library to the list, please drop a comment using the feedback link below.
+
+```
+
+### Libraries that define relations
+
+The following libraries provide programmatic instructions for relating to a specific charm.
+
+| Library | Used in | Description |
+| --- | --- | --- |
```

Comment:
We can make tables even less noisy with:

```
Col | Col | Col
-|-|-
cell | cell | cell
```

The outer delimiters are optional.

Reply:
Sorry, I pointed you to the wrong file here (the deleted markdown one, not the new rst one). This is fixed with a [list-table](https://github.com/canonical/charmcraft/pull/2010/files#diff-28bae703b56662486de50e2c07ee4f3a89de783d46374a673f46081cdef9d8a4) in the markdown version (that also matches the style guide).